File: 2369.py
def solution(N, A):
    counters = [0] * N
    max_counter = 0
    last_max_counter = 0
    
    for operation in A:
        if 1 <= operation <= N:
            # Increase counter X by 1
            if counters[operation - 1] < last_max_counter:
                counters[operation - 1] = last_max_counter
            
            counters[operation - 1] += 1
            
            if counters[operation - 1] > max_counter:
                max_counter = counters[operation - 1]
                
        elif operation == N + 1:
            # Set all counters to max_counter
            last_max_counter = max_counter
            
    for i in range(N):
        if counters[i] < last_max_counter:
            counters[i] = last_max_counter
            
    return counters


#print(solution(5, [1,1,4,6,1,5,5]))
import math
def procesar_estudiante(nombre, notas):
    print("El nombre del estudiante es", nombre)

    # Calcula el promedio
    promedio = math.floor(sum(notas) / len(notas))*10
    print(f"Su promedio es {promedio}/100")

    # Obtiene la nota más alta y la más baja
    nota_mas_alta = math.floor(max(notas))*10
    nota_mas_baja = math.floor(min(notas))*10
    print(f"Su nota mas baja es {nota_mas_baja}/100")
    print(f"Su nota mas alta es {nota_mas_alta}/100")

    # Determina si el estudiante está aprobado o reprobado
    if promedio >= 60:
        print("El estudiante esta aprobado")
    else:
        print("El estudiante esta reprobado")

    print()

casos = int(input())
for _ in range(casos):
    nombre = input()
    notas = list(map(int, input().split()))
    procesar_estudiante(nombre, notas)

File: jvCPP/2369.cpp
/*#include <iostream>;
using namespace std;

int main(){
    int t;
    cin >> t;
    while(t--){
        char str[10000];
        cin >> str;

    }
    return 0;
}*/
/*
#include <iostream>
#include <sstream>
#include <vector>

using namespace std;

int main() {
    int t;
    cin >> t;
    cin.clear();
	cin.clear();
    while(t--){
        string name;
        getline(cin, name);
        cin.clear();
	    cin.clear();
        string input;
        getline(cin, input);

        stringstream ss(input);
        vector<int> numbers;
        int num;
        while (ss >> num) {
            numbers.push_back(num);
        }

        // Imprimir los números leídos
        int sum =0;
        for (int n : numbers) {
            //cout << n << " ";
            sum = sum+ n;
        }
        cout << "El nombre del estudiante es "<<name<< endl;
        cout << "Su promedio es " <<(sum / numbers.size())*10 <<"/100";
        cout << endl;
    }
    

    return 0;
}
*/
#include <iostream>
#include <vector>
#include <algorithm>
#include <iomanip>

struct Estudiante {
    std::string nombre;
    std::vector<int> notas;
};

void procesarEstudiante(const Estudiante& estudiante) {
    std::cout << "El nombre del estudiante es " << estudiante.nombre << std::endl;

    // Calcula el promedio
    double promedio = 0.0;
    for (size_t i = 0; i < estudiante.notas.size(); ++i) {
        promedio += estudiante.notas[i];
    }
    promedio /= estudiante.notas.size();

    std::cout << "Su promedio es " << std::fixed << std::setprecision(0) << promedio << "/100" << std::endl;

    // Obtiene la nota más alta y la más baja
    int notaMasAlta = *std::max_element(estudiante.notas.begin(), estudiante.notas.end());
    int notaMasBaja = *std::min_element(estudiante.notas.begin(), estudiante.notas.end());

    std::cout << "Su nota mas baja es " << notaMasBaja << "/100" << std::endl;
    std::cout << "Su nota mas alta es " << notaMasAlta << "/100" << std::endl;

    // Determina si el estudiante está aprobado o reprobado
    if (promedio >= 60) {
        std::cout << "El estudiante esta aprobado" << std::endl;
    } else {
        std::cout << "El estudiante esta reprobado" << std::endl;
    }

    std::cout << std::endl;
}

int main() {
    int casos;
    std::cin >> casos;

    std::vector<Estudiante> estudiantes(casos);

    for (int i = 0; i < casos; ++i) {
        std::cin.ignore(); // Ignorar el salto de línea después de casos
        std::getline(std::cin, estudiantes[i].nombre);

        int cantidadNotas;
        std::cin >> cantidadNotas;

        estudiantes[i].notas.resize(cantidadNotas);
        for (int j = 0; j < cantidadNotas; ++j) {
            std::cin >> estudiantes[i].notas[j];
        }
    }

    for (int i = 0; i < casos; ++i) {
        procesarEstudiante(estudiantes[i]);
    }

    return 0;
}
